build-patched-kao-kcmo-images.sh

```diff
@@ -56,7 +56,7 @@ fi
 
 function release_image_for_arch() {
      local arch=$1
-     local mirror="https://mirror.openshift.com/pub/openshift-v4/${arch}/clients/ocp"
+     local mirror=$(echo ${MIRROR} | sed "s/$ARCH/$arch/g")
```

Comment:
Can you add a `mirror_for_arch` helper instead, I think it would be a lot easier to understand what it's doing.

Reply:
you mean something like this?
```bash
function release_image_for_arch() {
     local arch=$1
     local mirror=mirror_for_arch
     curl -L "${mirror}/${OPENSHIFT_RELEASE_VERSION}/release.txt" 2>/dev/null| sed -n 's/^Pull From: //p'
}

function mirror_for_arch() {
    echo ${MIRROR} | sed "s/$ARCH/$arch/g"
}
```

Reply:
```
diff --git a/build-patched-kao-kcmo-images.sh b/build-patched-kao-kcmo-images.sh
index 8d0358a..be372ab 100755
--- a/build-patched-kao-kcmo-images.sh
+++ b/build-patched-kao-kcmo-images.sh
@@ -35,10 +35,15 @@ function check_pull_secret() {
         fi
 }
 
+function mirror_for_arch() {
+	local arch=$1
+	echo "https://mirror.openshift.com/pub/openshift-v4/$arch/clients/ocp"
+}
+
 check_pull_secret
 
 ARCH=$(uname -m)
-MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/$ARCH/clients/ocp}
+MIRROR=$(mirror_for_arch $ARCH)
 
 # If user defined the OPENSHIFT_VERSION environment variable then use it.
 if test -n "${OPENSHIFT_VERSION-}"; then
@@ -56,7 +61,7 @@ fi
 
 function release_image_for_arch() {
      local arch=$1
-     local mirror="https://mirror.openshift.com/pub/openshift-v4/${arch}/clients/ocp"
+     local mirror=$(mirror_for_arch $arch)
      curl -L "${mirror}/${OPENSHIFT_RELEASE_VERSION}/release.txt" 2>/dev/null| sed -n 's/^Pull From: //p'
 }

 ```

Reply:
@cfergeau this patch remove the `MIRROR` env so in the downstream CI, I can choose MIRROR to `ocp-dev-preview` and `ocp` when creating the patch images it will be then not possible.

Reply:
But `release_image_for_arch` already did not use `MIRROR` from the environment, is this PR fixing an issue related to this?

Reply:
@cfergeau we want to consume `MIRROR` on both the place so it we can avoid duplication and also able to use different URL by using this `MIRROR` env.